### ISSUE TITLE
Make carbon mgt console & soap services disabled by default

### ISSUE THREAD

**Issue by darshanasbg** [wso2/product-is#20755]
**Is your suggestion related to an experience ? Please describe.**
Carbon Management Console has been deprecated with IS 7.0.0
https://is.docs.wso2.com/en/latest/references/about-this-release/#deprecated-features

SOAP Services has been deprecated since IS 6.0.0
https://is.docs.wso2.com/en/6.1.0/references/about-this-release/#deprecated-features

As the first step retiring these functionality, its better to disable these two in the default pack.

**Describe the improvement**
Have soap services and the carbon management console not accessible in the default pack.

Need to consider giving a feature flag to enable soap services to keep the backward compatibility for anyone who is migrating from an old version.

**Additional context**
Fully retiring these features would possibly be done with next major IS version.

**Comment by janakamarasena** [comment]
Would be good if soap services can be selectively enabled so anyone can enable only the ones they require.

**Comment by darshanasbg** [comment]
> Would be good if soap services can be selectively enabled so anyone can enable only the ones they require.

Yeah.. We can consider that..

**Comment by shashimalcse** [comment]
Identified a couple of SOAP services used in IS7 management console features related to API access management and authroization. Giving a feature flag to enable these features would be ideal. 

- Entitlement (XACML)
  - EntitlementAdminService (PDP Configurations)
  - EntitlementService (PDP Endpoint)
  - EntitlementPolicyAdminService (PAP)
- Roles List (Managing permissions of old roles. mail : Revamping Management Console in IS 7.0.0
)
  - UserAdmin (role management)
- Consent Purpose (Don't know the exact SOAP service used for this)

The following are some SOAP services not used in the console (We can disable these for the default pack):

- OAuth2TokenValidationService (token validation)
- ServerRolesManager (manage server roles)
- OAuthAdminService (manage oauth apps)
- IdentityApplicationManagementService (application mgt)
- ResourceAdminService

**Comment by darshanasbg** [comment]
Thanks for the analysis @shashimalcse..

IMO, we should not ask to access management console or soap services for any features that are not deprecated in the product.. We should have corresponding REST APIs and new UIs to use them..

We may allow selective access to soap services, considering the backward compatibility of the legacy deprecated features.. But that should be only used for that purpose..

**Comment by shashimalcse** [comment]
@darshanasbg agree! there is an effort ongoing with XACML REST APIs and the new console UI. Let's track those.

**Comment by darshanasbg** [comment]
Carbon products have a inbuilt capability to avoid registering the axis services that come through the OSGi bundles by setting a system property named `optimize`.
Ref: https://github.com/wso2/carbon-kernel/blob/6b7b18a83a551227bcda7484cf3735eb60dca52c/core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/init/CarbonServerManager.java#L482

For example if we start the product using `sh wso2server.sh -Doptimize=true`, it will not register the soap services.

Although this does not have the capability to enable services selectively, IMO we should use this capability and have the all the soap services disabled as the default functionality of the product as the starting point giving the benefit of reduced exposed surface for the majority of the product users..

**Comment by darshanasbg** [comment]
> Carbon products have a inbuilt capability to avoid registering the axis services that come through the OSGi bundles by setting a system property named `optimize`. Ref: https://github.com/wso2/carbon-kernel/blob/6b7b18a83a551227bcda7484cf3735eb60dca52c/core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/init/CarbonServerManager.java#L482
> 
> For example if we start the product using `sh wso2server.sh -Doptimize=true`, it will not register the soap services.
> 
> Although this does not have the capability to enable services selectively, IMO we should use this capability and have the all the soap services disabled as the default functionality of the product as the starting point giving the benefit of reduced exposed surface for the majority of the product users..

Implemented above with: <TDB>

**Comment by darshanasbg** [comment]
Disabling soap services restrict the use of the management console, thus management console landing page updated to look like following , if the soap services are disabled.. Note the header banner is updated and the sign-in form has been removed.

<img width="1502" alt="Image" src="https://github.com/user-attachments/assets/266a4790-61af-4a15-bd55-4ca7091b62af" />

**Comment by darshanasbg** [comment]
In summary, with the changes
- https://github.com/wso2/product-is/pull/23158
- https://github.com/wso2/carbon-kernel/pull/4242

it will make the management console and the soap services disabled by default and in order to make them available, product needs to start with the system param`-Doptimize=false` added.

ex:
```
sh wso2server.sh -Doptimize=true
```

**Comment by Sachin-Mamoru** [comment]
### Remaining work

PR [1] is ready to be merged. The following tasks need to be completed subsequently:

Test this - https://github.com/wso2/product-is/issues/20755#issuecomment-2668094624

**Product Documentation:**
Update the product documentation to include the new parameter, as described in the issue comment [here](https://github.com/wso2/product-is/issues/20755#issuecomment-2668266064).

**Connector README Updates:**
Update the README files of the respective connectors that use SOAP, reflecting the inclusion of this new parameter.

**Migration Documentation for 7.2.0:**
Update the migration guide for WSO2 Identity Server 7.2.0 to document this change.

Reference:
[1] https://github.com/wso2/product-is/pull/23158

**Comment by inthirakumaaran** [comment]
### Server Level Config to disable carbon mgt console and soap(admin) services

In the previous improvement, we used `-Doptimize=true` to disable the admin services. With this fix, we are introducing a new server-level configuration to achieve the same

```
[server]
...
disable_admin_services = false
```
From now on, both this configuration and the `-Doptimize=true` system property will be honored.

#### PRs

1. Main: 
      -  https://github.com/wso2/carbon-kernel/pull/4361
      - https://github.com/wso2/product-is/pull/25365
2. Test fix:(we need to enable the soap services for the integration tests)
      - https://github.com/wso2/carbon-platform-integration/pull/105
3. Backward compatibility (IS 7.0.0 and IS 7.1.0)
      - https://github.com/wso2/carbon-identity-framework/pull/7217

**Comment by inthirakumaaran** [comment]
The configuration name will be changed to below in the toml level to enable the soap services.

```
[server]
...
enable_admin_services = true
```
